### PR TITLE
testing torch==1.13.1+cu117 and torchvision==0.14.1+cu117 on 3.10

### PR DESCRIPTION
**Pull request by petalas**
Disclaimer: I don't really know enough about this.

Just thought I'd give it a go in case this is helpful in any way, if not feel free to close this PR.

The build I added [seems to have completed successfully on my test branch](https://github.com/petalas/xfromers_builds/actions/runs/3761898149/jobs/6394076435).

Interestingly 2 of the existing ones failed? [This one](https://github.com/petalas/xfromers_builds/actions/runs/3761898149/jobs/6394076196) and [this one](https://github.com/petalas/xfromers_builds/actions/runs/3761898149/jobs/6394076246).

Is this how you'd create a cu117 build for python 3.10?
Hoping we can use it with Automatic1111's webui.

### CODE REVIEW

**Comment by ninele7** [comment]
Hey, this repo was made as proof of concept, and contains some dirty hack to make it working. Also I'm not really following latest progress and don't post updates. 

Recently xformers developers accepted PR with action to build different wheel, that maintained lots batter than mine: https://github.com/facebookresearch/xformers/blob/main/.github/workflows/wheels.yml. As far as I see they support pytorch version that you need, but don't use PTX (might be needed to get more performance on 4090). 

You can try their wheels from here: https://github.com/facebookresearch/xformers/actions/runs/3758583159 or from here: https://pypi.org/project/xformers/#history (pre-release ones).

Also one of developers in a1111 repo said this: https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/3876#discussioncomment-4076937. Lovelace is 4090 architecture. So if you need max performance on 4090 you might need to wait more for proper support in pytorch and xformers.

TLDR: If I was you I would fork xformers repo, changed cuda version to one more modern like 11.8 (it supports 4090 better), tried to apply hack proposed in discussion above if it didn't work, I would try to enable PTX in their cuda arch list and use wheel built by this action.

I'm not closing it for now, because in following 2-3 weeks I might find time to catch up with all info about building xformers and to think if this repo is still relevant and how it can be relevant.

Feel free to ask additional questions.

**Comment by petalas** [comment]
@ninele7 thanks for all the info, when I have more time I might try to build with cuda 11.8 (this is all new to me, might be too much effort vs someone who already knows what they're doing). 

Anyway for now I've updated [my a1111 PR](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/5939) to use:
`torch==1.13.1+cu117 torchvision==0.14.1+cu117 xformers==0.0.16rc395`

No performance improvement with this PR by itself unfortunately but hopefully a step in the right direction.

**Comment by ninele7** [comment]
If you don' need all features of a1111 repo, you can try to use https://github.com/VoltaML/voltaML-fast-stable-diffusion (requires docker). You can check performance metrics in bottom part of readme. As far as I know they currently working on upstreaming they results to popular frontends but it might be blocked by lack of native windows support.

**Comment by aliencaocao** [comment]
I have been using xformers on cuda 11.8 since this repo existed, with cu117 torch packages. No issues so far.
